packages/core/src/rum/instrumentation/resourceTracking/domain/distributedTracing.tsx

```diff
@@ -29,61 +33,92 @@ const DISCARDED_TRACE_ATTRIBUTES: DdRumResourceTracingAttributes = {
+        traceId: new TraceIdentifier() as TraceId,
+        spanId: new TraceIdentifier() as SpanId,
+        samplingPriorityHeader: isSampled ? '1' : '0',
+        tracingStrategy: 'KEEP',
+        rulePsr: tracingSamplingRate / 100,
+        propagators
+    };
+
+    return tracingAttributes;
 };
 
-/*
- * This code was inspired from browser-sdk at (https://github.com/DataDog/browser-sdk/blob/master/packages/rum-core/src/domain/tracing/tracer.ts#L107)
+/**
+ * Using branded types will ensure we don't accidentally use
+ * traceId for spanId when generating headers.
  */
-export function generateTraceId(): string {
-    const radix = 10;
-    const MAX_32_BITS_NUMBER = 4294967295; // 2^32-1
-    let low: number = Math.floor(Math.random() * MAX_32_BITS_NUMBER);
-    let high: number = Math.floor(Math.random() * MAX_32_BITS_NUMBER);
-    let str = '';
+export type TraceId = TraceIdentifier & {
+    _brand: 'traceId';
+};
 
-    while (high > 0 && low > 0) {
-        // Create an intermediate value with the same modulo as the combined high and low value
-        // but requiring 36 bits max (32 for the low value + 4 for the high part)
-        const modH = high % radix;
-        const temp = (modH << 32) + low;
-        const digit = temp % radix;
+export type SpanId = TraceIdentifier & {
+    _brand: 'spanId';
+};
 
-        // update the high value
-        high = (high - modH) / radix; // we reuse the modH to avoid the need of a floor op
-        // the low value reuses the previous temp value to account for the "missing mod" in the high update
-        low = (temp - digit) / radix; // we reuse the digit to avoid the need of a floor op
+/*
+ * This code was inspired from browser-sdk at (https://github.com/DataDog/browser-sdk/blob/0e9722d5b06f6d49264bc82cd254a207d647d66c/packages/rum-core/src/domain/tracing/tracer.ts#L190)
+ */
+const MAX_32_BITS_NUMBER = 4294967295; // 2^32-1
+const MAX_31_BITS_NUMBER = 2147483647; // 2^31-1
+export class TraceIdentifier {
+    private low: number;
+    private high: number;
 
-        // update the string from right to left
-        str = digit.toString() + str;
+    constructor() {
+        // We need to have a 63 bits number max
+        this.high = Math.floor(Math.random() * MAX_31_BITS_NUMBER);
+        this.low = Math.floor(Math.random() * MAX_32_BITS_NUMBER);
     }
-    return str;
+
+    toString = (radix: number) => {
+        let low = this.low;
+        let high = this.high;
+        let str = '';
+
+        while (high > 0 || low > 0) {
+            const mod = (high % radix) * (MAX_32_BITS_NUMBER + 1) + low;
+            high = Math.floor(high / radix);
+            low = Math.floor(mod / radix);
+            str = (mod % radix).toString(radix) + str;
+        }
+        return str;
+    };
+
+    toPaddedString = (radix: number, length: number) => {
+        const traceId = this.toString(radix);
+        if (traceId.length > length) {
+            return traceId;
```

Comment:
Should we explain in the JSDoc that `length` is the lower boundary and not the exact length returned? Otherwise (if if should be the exact length returned), we should truncate here?